### ISSUE TITLE
issue:  Invoke setValue(`shema.list.${idx}`, object) after an asynchronous is not work

### ISSUE THREAD

**Issue by hieusmiths** [react-hook-form/react-hook-form#10709]
### Version Number

7.45.1

My previous verion:  7.24.0 and everything working well

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-validationschema-forked-g97nmt?file=/package.json

### Steps to reproduce

- We having an array field look like bellow. In that contain Editor of DraftJS

```tsx
label: string;
content: EditorState (Data type of DraftJS);
etc...

```

- Click a button to call data from the server
- Calculate that data and set the value corresponding to the array field

The following are three ways I tried to check the 7.45.1 working or not. 

```tsx
const onSetValAsync = async (idx = 0) => {
    await delay(500);
    setValue(`shema.list.${idx}`, {
      obj: {
        content: createEditorStateFromText(Math.random().toString()),
        label: Math.random().toString()
      }
    });
  };
 
  - 7.24.0: Good
  - 7.45.1: Only update the label field
  ```
  
  ```tsx
  const onSetValSync = async (idx = 0) => {
    setValue(`shema.list.${idx}`, {
      obj: {
        content: createEditorStateFromText(Math.random().toString()),
        label: Math.random().toString()
      }
    });
  };
  
  - 7.24.0: Good
  - 7.45.1: Only update the label field
```

```tsx
  const onSetValSingleTargetAsync = async (idx = 0) => {
    await delay(500);
    setValue(
      `shema.list.${idx}.obj.content`,
      createEditorStateFromText(Math.random().toString())
    );

    setValue(`shema.list.${idx}.obj.label`, Math.random().toString());
    
   - 7.24.0: Good
   - 7.45.1: Good but in my local it's not working and 
    we can't use this way for our's behavior (We have a lot fields need to update)
  };
```

My solution is move setValue into another place. Base on the rule don't stack action
In this case, I use `useEffect`

Could you help me take a look and explain why it can work? Thanks 

```tsx
const [data, setData] -= useState(null)
const onSetValAsync = async (idx = 0) => {
    await delay(500);
  
     setData({
        content: Math.random().toString(),
        label: Math.random().toString()
        etc...
     })
    
  };
  
  useEffect(() => {
  if (data) {
      setValue(`shema.list.${idx}`, {
        obj: {
          content: createEditorStateFromText(data.content),
          label: data.label,
          etc...
        }
      });
  }
  }, [data])
```

### Expected behaviour

setValue working well.

NOTE: this works perfectly with version 7.24.0 -> 7.25.0. I'm using 7.24.0 

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> NOTE: this works perfectly with version 7.24.0 -> 7.25.0. I'm using 7.24.0

why version stopped working?

**Comment by dwiyatci** [comment]
@bluebill1049 Actually, we got a similar issue in our app after upgrading from 7.45.0 to 7.45.1; in our case, it occurred with a simple value (not even a complex object) update:
```js
sleep(500); // async op
setValue('foo.bar.baz', 42); // failed in 7.45.1 but worked in 7.45.0

function sleep(time) {
  return new Promise((resolve) => setTimeout(resolve, time));
}
```

**Comment by bluebill1049** [comment]
> @bluebill1049 Actually, we got a similar issue in our app after upgrading from 7.45.0 to 7.45.1; in our case, it occurred with a simple value (not even a complex object) update:
> 
> ```js
> await sleep(500); // async op
> setValue('foo.bar.baz', 42); // failed in >= 7.45.1 but worked in 7.45.0
> 
> function sleep(time) {
>   return new Promise((resolve) => setTimeout(resolve, time));
> }
> ```

Please create a simple codesandbox, create a codesnipt doesn't really help too much here. thanks

**Comment by dwiyatci** [comment]
> Please create a simple codesandbox

Yeah, I'm sorry. 🙏 Been trying for hours, but unfortunately I didn't manage to contain my issue this time. 😔  And probably it doesn't even have anything to do with async op (it was a misleading debugging).

All I know now is that ``setValue(`items.${index}.x`, value)`` started failing in v7.45.1, and if I revert this commit (https://github.com/react-hook-form/react-hook-form/pull/10586/files) in the lib source code, it starts working again. That commit ironically fixed my other issue (https://github.com/react-hook-form/react-hook-form/issues/10585). I guess I'll downgrade to v7.45.0 for now with a workaround for that issue. Sorry again that I cannot give you more leads here...

**Comment by dwiyatci** [comment]
@bluebill1049 Ah, but I saw that you have reverted it already on `master` 2 days ago? 👉 https://github.com/react-hook-form/react-hook-form/commit/3ca17019c86535a331e32b465019ec25bbbf02fb

**Comment by bluebill1049** [comment]
oh @dwiyatci that was the root cause?

**Comment by dwiyatci** [comment]
Yep, our app seems to be back working normally again with the revert

**Comment by bluebill1049** [comment]
thanks @dwiyatci i will close this and release the patch this week. Thanks for confirming.

**Comment by bluebill1049** [comment]
> @bluebill1049 Thanks ever so much! But do you think there's still a chance to fix #10585 along with the revert?

I doubt it, I have tried to address but it brought more regession.
